ui/component/or-rules/src/or-rule-tree.ts

```diff
@@ -609,4 +629,23 @@ export class OrRuleTree extends OrTreeMenu {
                 return "iconfill-gray";
         }
     }
+
+    /**
+     * Programmatically finds a group by its ID and sets it to be expanded.
+     * This is called by the parent component after a new rule is added to a group.
+     * @param {string} groupId The ID of the group to expand.
+     */
+    public expandGroup(groupId: string) {
+        // Wait for any pending Lit updates to complete to ensure the node list is stable.
+        this.updateComplete.then(() => {
+            const groupNode = this.nodes.find(node => node.id === groupId && node.children);
+
+            if (groupNode) {
+                groupNode.expanded = true;
+                // re-render with the change.
+                this.requestUpdate();
+            }
+        });
+    }
```

Comment:
Can you move this to the base class of `or-tree-menu` ?
So the function is available for every tree menu in the UI 👍